base/expr.jl

```diff
@@ -118,6 +118,20 @@ macro eval(x)
     :($(esc(:eval))($(Expr(:quote,x))))
 end
 
+"""
+    @inline
+
+Small functions typically don't even need the @inline annotation,
+it does it on its own. By using @inline on bigger functions,
+an extra nudge can be given to the compiler, like in the example below.
+```
+    @inline function bigfunction(x)
+        #=
+           Function Definition
+        =#
+    end
```

Comment:
Shall I open a new issue recommending the change you suggested? I would like to learn how to